### ISSUE TITLE
Конфиги на GO

### ISSUE THREAD

**Issue by MrmaderatorYT** [Distemi/ULE#2]
Глянул тутор на хабре и не понятно как мигрировал на Rust да и я шарю на Go. Есть ссылка на весь проект на Go языке?
Комиты глянул - всё на расте

**Comment by Distemi** [comment]
Данный проект, написанный на Go к сожалению утерян, но вы всегда можете посмотреть [официальную библиотеку](https://github.com/Tnze/go-mc), которая использовалась, там можно найти код подробнее в папке examples.

Проект мигрировал на Rust после того, как разбирая создания программ на Go заметил небольшие трудности в создании больших проектов, конечно при корректной архитектуре получалось бы всё сделать на Go также, но всё равно мне раст приятнее оказался. На момент написания этого кода я только начинал углубляться в те языки(Rust, Go), поэтому вот...

**Comment by MrmaderatorYT** [comment]
можно тогда я на хабре продолжу тему ядра на основе вашего кода?
Что на счёт доквы - там по уберали протоколы и мне пришлось знатно поискать файлы.
Также переделывать пару файлов из-за того что поудаляли файлы с гитхаба которые были импортированы

**Comment by Distemi** [comment]
Конечно, можете взять тот код, по крайней мере там и так мало написано для полноценного функционала.

**Comment by MrmaderatorYT** [comment]
хочу спросить. а что означает ULE?
вот в чём смысл этого названия?)

**Comment by Distemi** [comment]
Сам точно не припомню, но вроде Unique Launched Edition возможно 😅, сам в локальном проекте ядра уже по другому называю.

**Comment by MrmaderatorYT** [comment]
Нету файла генерации мира?
а то как- то сложно реализовать. Что-то сделал на Go не думаю
`use std::fs::File;
use std::io::{Cursor, Write};
use byteorder::{BigEndian, WriteBytesExt};

fn main() -> std::io::Result<()> {
    let mut file = File::create("./r.0.0.mca")?;

    for chunk_x in 0..32 {
        for chunk_z in 0..32 {
            let mut chunk_data: Vec<u8> = vec![0; 32768];

            for y in 0..128 {
                for z in 0..16 {
                    for x in 0..16 {
                        let index = ((y * 16 + z) * 16 + x) as usize;
                        let block_id = if y == 0 { 7 } else { 0 };
                        chunk_data[index] = block_id;
                    }
                }
            }

            let mut chunk_header: Vec<u8> = vec![0; 5];
            chunk_header.write_i32::<BigEndian>(chunk_x)?;
            chunk_header.write_i32::<BigEndian>(chunk_z)?;
            let mut cursor = Cursor::new(&mut chunk_header[4..]);
            cursor.write_u8(1)?;

            file.write_all(&chunk_header)?;
            file.write_all(&chunk_data)?;
        }
    }

    Ok(())
}`

Вот я на расте чёт сделал. Если нужно...

**Comment by Distemi** [comment]
К сожелению я пока в данном репозитории не делал генерации и отправки мира(как и входа в принципе).

За код на расте спасибо)
Хотя я думаю лучше использовать STD или Крейт bytes для такого. Правда пока-что не делал работу с мирами полноценную, но всё равно спасибо)
